frontend/frontend/src/sidebar/Sidebar.tsx

```
@@ -0,0 +1,226 @@
+import React, { useEffect, useRef, useState } from "react";
+import { useLocation, NavLink } from "react-router-dom";
+import { getAppMenus } from "../services/app-menus";
+import type { MenuItem } from "./MenuItem";
+import type { SidebarProps } from "./SidebarProps";
+import "./sidebar.css";
+
+const LS_KEY = "app_sidebar_minified";
+
+const Sidebar: React.FC<SidebarProps> = ({ appSidebarMinified = false }) => {
+  const location = useLocation();
+
+  const [menus] = useState<MenuItem[]>(() => getAppMenus());
+
+  const readInitialMinified = (): boolean => {
+    try {
+      const raw = localStorage.getItem(LS_KEY);
+      if (raw === "true") return true;
+      if (raw === "false") return false;
+    } catch {}
+    return Boolean(appSidebarMinified);
+  };
+
+  const mountedRef = useRef(false);
+
+  const [isMinified, setIsMinified] = useState<boolean>(() => readInitialMinified());
+
+  const getInitialExpanded = (): string | null => {
+    const activeParent = getAppMenus()
+      .flatMap((section) => section.submenu ?? [])
+      .find((item) => item.submenu?.some((sub) => location.pathname.startsWith(sub.url ?? "")));
+
+    return activeParent ? activeParent.title : null;
+  };
+
+  const [expanded, setExpanded] = useState<string | null>(() => getInitialExpanded());
+
+  const [hoveredMenu, setHoveredMenu] = useState<MenuItem | null>(null);
+  const [hoverPosition, setHoverPosition] = useState<number>(0);
+  const hoverTimeout = useRef<number | null>(null);
+
+  useEffect(() => {
+    if (!mountedRef.current) {
+      mountedRef.current = true;
+      return;
+    }
+    try {
+      localStorage.setItem(LS_KEY, isMinified ? "true" : "false");
+    } catch {}
```

just let the exception be thrown instead of silently sweeping it under the rug